src/utils.h

```diff
@@ -31,7 +38,11 @@ extern bool ts_type_is_int8_binary_compatible(Oid sourcetype);
  * Will throw an error for that, or other conversion issues.
  */
 extern TSDLLEXPORT int64 ts_time_value_to_internal(Datum time_val, Oid type);
+extern int64 ts_time_value_to_internal_or_infinite(Datum time_val, Oid type_oid,
```

Comment:
why do we need two variants of this function? shouldn't callers always be using the ts_time_value_to_internal_or_infinite variant.Is that change for another PR?

Reply:
we should not really store the result of `ts_time_value_to_internal_or_infinite` since most postgres functions cannot handle infinities, it's really mostly useful for our own computation, where we know how to handle it.